src/Analysis/Ast/Impl/Extensions/PythonModuleExtensions.cs

```diff
@@ -26,5 +28,23 @@ internal static PythonAst GetAst(this IPythonModule module)
             => ((IAstNodeContainer)module).GetAstNode<T>(o);
         internal static void AddAstNode(this IPythonModule module, object o, Node n)
             => ((IAstNodeContainer)module).AddAstNode(o, n);
+
+        /// <summary>
+        /// Returns the string line corresponding to the given location
+        /// </summary>
+        /// <param name="line">The line number</param>
+        internal static string GetLine(this PythonModule module, int line) {
```

Comment:
During analysis AST attached to the module analysis is old. `module.Analysis` is actually previous. You need to explicitly use AST that is in the current evaluator.

Reply:
I have observed doing GetAst(module) gets the most current AST.